src/symbol/collision_tile.js

```diff
@@ -150,6 +150,16 @@ class CollisionTile {
             box.bbox2 = x2;
             box.bbox3 = y2;
 
+            // When the map is pitched the distance covered by a line changes.
+            // Adjust the max scale by (approximatePitchedLength / approximateRegularLength)
+            // to compensate for this.
+            const yStretch2 = yStretch;
+            const xSqr = box.offsetX * box.offsetX;
```

Comment:
Don't these offsets need to have a rotation applied to them? I think as is this only works for labels that are relatively horizontal relative to tile orientation.